### PR TITLE
Added example and code changes to support registry as service center for sidecar injector

### PR DESCRIPTION
**Pull request by savitaashture**

### CODE REVIEW

File: webhook/webhook.go [hunk 1]
```diff
@@ -35,6 +35,11 @@ var (
 const (
 	webhookInjectKey = "sidecar-injector-mesher.io/inject"
 	webhookStatusKey = "sidecar-injector-mesher.io/status"
+	mesherServer     = "sidecar.mesher.io/servicePorts"
```
Comment (tianxiaoliang) [inline webhook/webhook.go]: use same key sidecar.mesher.io

Comment (tianxiaoliang) [inline webhook/webhook.go]: mesherServer     is not clear for var name, use servicePortsAnnotationKey

File: webhook/webhook.go [hunk 2]
```diff
@@ -292,6 +312,24 @@ func (wh *WebHookServer) mutation(ar *v1beta1.AdmissionReview) *v1beta1.Admissio
...
+	// This sidecar container will be common to consumer and provider
+	// So before proceeding first check the env "SERVICE_PORTS". If its present delete the env
+	for k := range wh.SidecarConfig.Containers {
+		for i := range wh.SidecarConfig.Containers[k].Env {
+			if wh.SidecarConfig.Containers[k].Env[i].Name == servicePorts {
+				wh.SidecarConfig.Containers[k].Env = append(wh.SidecarConfig.Containers[k].Env[0:i], wh.SidecarConfig.Containers[k].Env[i+1:]...)
+			}
+		}
+	}
+
```
Comment (tianxiaoliang) [inline webhook/webhook.go]: add comment //TODO read service endpoints if there is , it has then use it  as service ports

Comment (savitaashture) [reply webhook/webhook.go]: Added comment

**Pull request merged**